src/tool/hpcrun/gpu/gpu-activity.h

```diff
@@ -325,6 +326,13 @@ typedef struct gpu_branch_t {
 } gpu_branch_t;
 
 
+typedef struct gpu_blame_shift_t {
+  double cpu_idle_time;
+  double gpu_idle_time;
+  double cpu_idle_cause_time;
```

Comment:
What is `cpu_idle_cause_time` and why there's no `gpu_idle_cause_time`?

Reply:
**how is cpu_idleness detected?**
call to sync() makes the cpu thread wait. In this case we know who is waiting(cpu thread) and who is the cause for wait(either the gpu stream or a kernel inside the stream). Thus we can increment both cpu_idle time and cpu_idle_cause appropriately.

**How is gpu idleness detected?** 
itimer callbacks check the number of active kernels periodically. When (#kernels == 0), the gpu_idle_time is incremented. Note that the stacktrace where the metric is recorded is responsible for the gpu idleness. gpu_idle_cause_time does not make sense. Where would we attribute that metric? And how will it be calculated?

Reply:
By reviewing Milind's paper, I reminded what `cpu_idle_cause` means. In fact, the original paper denotes `gpu_idle_time` as `gpu_idle_cause_time`.

Reply:
done